### ISSUE TITLE
[core] signal required in requests.

### ISSUE THREAD

**Issue by xPolar** [discordjs/discord.js#9369]
### Which package is this bug report for?

core

### Issue description

Make a request, such as the following examples, `this.client.api.guilds.get(interaction.guild_id)`, `this.client.api.interactions.defer(interaction.id, interaction.token)`, `this.client.api.interactions.getOriginalReply(interaction.application_id, interaction.token);`

Get the following error type error.

```
lib/classes/ApplicationCommand.ts:164:49 - error TS2554: Expected 2 arguments, but got 1.

164      const guild = await this.client.api.guilds.get(interaction.guild_id);
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/.pnpm/@discordjs+core@0.5.0_bufferutil@4.0.7_utf-8-validate@6.0.3/node_modules/@discordjs/core/dist/files-4cadec53.d.ts:413:26
    413     get(guildId: string, { signal }: Pick<RequestData, 'signal'>): Promise<discord_api_types_v10.APIGuild>;
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument matching this binding pattern was not provided.
```

### Package version

@discordjs/core 0.5.0

### Node.js version

v18.12.1

### Operating system

MacOS 13.2.1

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

Not applicable (subpackage bug)

### Which gateway intents are you subscribing to?

Not applicable (subpackage bug)

**Comment by almeidx** [comment]
Fixed in #9311, but it's not in a stable release yet. You can use `@discordjs/core@dev` for the time being

**Comment by xPolar** [comment]
Got it, thanks!
